### PR TITLE
Homework 3

### PR DESCRIPTION
**Pull request by KirillKlimenok**
homework 3

### CODE REVIEW

File: homework3/task1/task1.cpp [hunk 1]
```diff
@@ -0,0 +1,36 @@
+﻿#include <iostream>
+
+using namespace std;
+
+double powerNumber(double number, double power);
+bool isCurrentPower(double number);
```
Comment (VadzimAlkhimionak) [inline homework3/task1/task1.cpp:7]: Great!

File: homework3/task1/task1.cpp [hunk 2]
```diff
@@ -0,0 +1,36 @@
+﻿#include <iostream>
+
+using namespace std;
+
+double powerNumber(double number, double power);
```
Comment (VadzimAlkhimionak) [inline homework3/task1/task1.cpp]: Your function returns some kind of result so give one name starting with a verb

File: homework3/task1/task1.cpp [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+bool isCurrentPower(double number);
+
+int main()
+{
+	double number, power;
+
+	cout << "Please, enter number: ";
+	cin >> number;
+	cout << "and enter power: ";
+	cin >> power;
```
Comment (VadzimAlkhimionak) [inline homework3/task1/task1.cpp]: Create function for this code

File: homework3/task1/task1.cpp [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+}
+
+double powerNumber(double number, double power) {
+	for (int i = 1; i < power; i++)	number *= number;
+
+	return number;
+}
+
+bool isCurrentPower(double power) {
+	return (double)((int)power) == power;
```
Comment (VadzimAlkhimionak) [inline homework3/task1/task1.cpp:35]: ?

File: homework3/task2/task2.cpp [hunk 5]
```diff
@@ -0,0 +1,41 @@
+﻿#include <iostream>
+
+using namespace std;
+
+void numbersWhoHaveUnitsInBinarySystem(int inputNumber);
```
Comment (VadzimAlkhimionak) [inline homework3/task2/task2.cpp]: so long name for function

File: homework3/task2/task2.cpp [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+
+int main()
+{
+	int inputNumber = 0;
+
+	cout << "Please, enter number: ";
+	cin >> inputNumber;
```
Comment (VadzimAlkhimionak) [inline homework3/task2/task2.cpp]: create function

File: homework3/task2/task2.cpp [hunk 7]
```diff
@@ -0,0 +1,41 @@
...
+		return -1;
+	}
+	numbersWhoHaveUnitsInBinarySystem(inputNumber);
+
+	return 0;
+}
+
+void numbersWhoHaveUnitsInBinarySystem(int inputNumber) {
+	int NumberWithUnits = 7;
+	int stepOfBinaryNumbers = 4;
```
Comment (VadzimAlkhimionak) [inline homework3/task2/task2.cpp:24]: 7 and 4 
What do you mean?

File: homework3/task2/task2.cpp [hunk 8]
```diff
@@ -0,0 +1,41 @@
...
+
+	if (inputNumber < 7) {
```
Comment (VadzimAlkhimionak) [inline homework3/task2/task2.cpp:26]: Why 7?

File: homework3/task2/task2.cpp [hunk 9]
```diff
@@ -0,0 +1,41 @@
...
+	else {
+		for (int i = 0; i <= inputNumber; i++) {
+			if (i == NumberWithUnits) {
+				cout << NumberWithUnits << '	';
+
+				NumberWithUnits = NumberWithUnits + stepOfBinaryNumbers * 2;
+				stepOfBinaryNumbers *= 2;
+			}
+			else {}
+		}
```
Comment (VadzimAlkhimionak) [inline homework3/task2/task2.cpp:38]: Explain it.
I guess you need to create function for this code

File: homework3/task3/task3.cpp [hunk 10]
```diff
@@ -0,0 +1,44 @@
+﻿#include <iostream>
+
+using namespace std;
+
+int fabonachiNumber(int number);
```
Comment (VadzimAlkhimionak) [inline homework3/task3/task3.cpp]: start with a verb

**Review by VadzimAlkhimionak — COMMENTED** [review]
Hello! I left some comments! Well done.
We will discuss some aspects in the lesson

File: homework3/task1/task1.cpp (hunk 4 above)
Comment (KirillKlimenok) [reply homework3/task1/task1.cpp:35]: Сhecking whether it is a fractional number or not

File: homework3/task1/task1.cpp (hunk 2 above)
Comment (KirillKlimenok) [reply homework3/task1/task1.cpp]: image

Comment (VadzimAlkhimionak) [reply homework3/task1/task1.cpp]: I think getPowerNumber more suitable

File: homework3/task1/task1.cpp (hunk 3 above)
Comment (KirillKlimenok) [reply homework3/task1/task1.cpp]: image
 done

File: homework3/task1/task1.cpp (hunk 2 above)
Comment (KirillKlimenok) [reply homework3/task1/task1.cpp]: done

File: homework3/task1/task1.cpp (hunk 3 above)
Comment (VadzimAlkhimionak) [reply homework3/task1/task1.cpp]: No, create one function instead of two one

File: homework3/task1/task1.cpp (hunk 4 above)
Comment (VadzimAlkhimionak) [reply homework3/task1/task1.cpp:35]: Ok

**Pull request merged**